content/articles/imbalanced-learn-to-handle-imbalanced-text-data-in-natural-language-processing/index.md

```diff
@@ -0,0 +1,638 @@
+
+We will first build a spam classifier model with natural language processing without balancing the classes in the dataset. We will implement the same model but use Imbalanced-Learn to balance the classes. Finally, we will compare the two models (before and after balancing) and evaluate their performance.
+
+### Table of contents
+- [Prerequisites](#prerequisites)
+- [Spam classification dataset](#spam-classification-dataset)
+- [Renaming the dataset column](#renaming-the-dataset-column)
+- [Calculating the length of each data sample](#calculating-the-length-of-each-data-sample)
+- [Text cleaning](#text-cleaning)
+- [Installing NLTK](#installing-nltk)
+- [Creating the custom functions](#creating-the-custom-functions)
+- [Converting the class labels into integer values](#converting-the-class-labels-into-interger-values)
+- [Implementing text vectorization](#implementing-text-vectorization)
+- [Splitting the vectorized dataset](#splitting-the-vectorized-dataset)
+- [Model building](#model-building)
+- [Making predictions using the test set](#making-predictions-using-the-test-set)
+- [Getting accuracy score of these predictions](#getting-accuracy-score-of-these-predictions)
+- [Implementing Imbalanced-learn](#implementing-imbalanced-learn)
+- [Splitting the text dataset](#splitting-the-text-dataset)
+- [Applying `RandomOverSampler` function](#applying-randomoversampler-function)
+- [Using the balanced dataset to build the same model](#using-the-balanced-dataset-to-build-the-same-model)
+- [Predicting using the model](#predicting-using-the-model)
+- [Getting accuracy score](#getting-accuracy-score)
+- [Conclusion](#conclusion)
+- [References](#references)
+
+### Prerequisites
+Ensure you know the following concepts to follow along easily and understand the tutorial:
+
+- Know [Python Programming] (https://www.w3schools.com/python/) concepts.
+- Know natural language concepts and how to [implemment an NLP model](/engineering-education/building-autocorrect-feature-using-nlp-with-python/).
+- Understand [machine learning modelling] (/engineering-education/house-price-prediction/) and other related concepts.
+- Implement the spam classification model in [Google Colab](https://research.google.com/colaboratory/) to get the same results.
+
+### Spam classification dataset
+We will use the SMS collection dataset to train the NLP model. It has two labeled classes. (`spam` and `ham`). 
+
+The `spam` class contains all the spam SMS. The `ham` class has all the SMS that are not spam. 
+
+The NLP model will classify an SMS as either spam or not spam. Ensure you can download the SMS collection dataset from [here](https://drive.google.com/file/d/1OLdz-fSUBEARnMqUdrTW4bkmGy8yUpCE/view?usp=sharing).
```

Comment:
As a rule of thumb, ensure that your paragraphs are at least three lines long before adding spaces. Fix this issue throughout your tutorial